src/tools/generate_workspace/src/main/java/com/google/devtools/build/workspace/maven/DefaultModelResolver.java

```diff
@@ -134,7 +135,12 @@ private UrlModelSource getModelSource(
 
   private boolean pomFileExists(URL url) {
     try {
-      HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+      URLConnection urlConnection = url.openConnection();
+      if (!(urlConnection instanceof HttpURLConnection)) {
```

Comment:
I saw only FtpURLConnection. But, anyway, it could return anything (depending on protocol in url), so we should be protective, and not fail with ClassCastException